### PR TITLE
DAP-03: move extensions to ciphertext.

### PR DESCRIPTION
**Pull request by branlwyd**
This commit introduces the PlaintextInputShare structure used for input shares in DAP-03.

Part of #761.

### CODE REVIEW

File: aggregator/src/aggregator.rs [hunk 1]
```diff
@@ -1249,6 +1249,8 @@ impl VdafOps {
         for<'a> &'a A::AggregateShare: Into<Vec<u8>>,
         C: Clock,
     {
+        // XXX: update
+
```
Comment (tgeoghegan) [inline aggregator/src/aggregator.rs:1253]:
```suggestion
```

File: messages/src/lib.rs [hunk 2]
```diff
@@ -853,31 +848,66 @@ impl ReportMetadata {
...
     fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
         let report_id = ReportId::decode(bytes)?;
         let time = Time::decode(bytes)?;
+
+        Ok(Self { report_id, time })
+    }
+}
+
+// XXX: documentation
+// XXX: tests
```
Comment (tgeoghegan) [inline messages/src/lib.rs:870]: Indeed, this needs doccomment and tests. Same goes on L877, L885, L890.

**Review by tgeoghegan — CHANGES_REQUESTED** [review]
Just needs some comment cleanup and some tests.

File: messages/src/lib.rs (hunk 2 above)
Comment (branlwyd) [reply messages/src/lib.rs]: Argh, one day I will remember to `git grep XXX` consistently before sending a PR for review. Sorry about that!

**Review by divergentdave — APPROVED** [review]

File: messages/src/lib.rs (hunk 2 above)
Comment (tgeoghegan) [reply messages/src/lib.rs]: No worries, I do it all the time myself. Maybe we need some kind of git pre-push hook that does that grep for us.

**Review by tgeoghegan — APPROVED** [review]

File: aggregator/src/aggregator.rs [hunk 3]
```diff
@@ -1519,13 +1523,21 @@ impl VdafOps {
                 })
             });
 
+            let plaintext_input_share = plaintext.and_then(|plaintext| {
+                PlaintextInputShare::get_decoded(&plaintext).map_err(|error| {
+                    info!(task_id = %task.id(), metadata = ?report_share.metadata(), ?error, "Couldn't decode helper's plaintext input share");
+                    aggregate_step_failure_counter.add(&Context::current(), 1, &[KeyValue::new("type", "plaintext_input_share_decode_failure")]);
```
[the remark this answers is not in the document]
Comment (branlwyd) [reply aggregator/src/aggregator.rs:1529]: (I'm going to do this in the follow-on PR I'm about to send for review, to avoid needing to re-run CI.)

**Pull request merged**